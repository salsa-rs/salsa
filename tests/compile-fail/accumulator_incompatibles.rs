#[salsa::accumulator(returns(ref))]
struct AccWithRetRef(u32);

#[salsa::accumulator(specify)]
struct AccWithSpecify(u32);

#[salsa::accumulator(no_eq)]
struct AccWithNoEq(u32);

#[salsa::accumulator(data = MyAcc)]
struct AccWithData(u32);

#[salsa::accumulator(db = Db)]
struct AcWithcDb(u32);

#[salsa::accumulator(recover_fn = recover)]
struct AccWithRecover(u32);

#[salsa::accumulator(lru = 12)]
struct AccWithLru(u32);

#[salsa::accumulator(constructor = Constructor)]
struct AccWithConstructor(u32);

fn main() {}
