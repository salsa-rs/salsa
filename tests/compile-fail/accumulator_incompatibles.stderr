error: `returns` option not allowed here
 --> tests/compile-fail/accumulator_incompatibles.rs:1:22
  |
1 | #[salsa::accumulator(returns(ref))]
  |                      ^^^^^^^

error: `specify` option not allowed here
 --> tests/compile-fail/accumulator_incompatibles.rs:4:22
  |
4 | #[salsa::accumulator(specify)]
  |                      ^^^^^^^

error: `no_eq` option not allowed here
 --> tests/compile-fail/accumulator_incompatibles.rs:7:22
  |
7 | #[salsa::accumulator(no_eq)]
  |                      ^^^^^

error: `data` option not allowed here
  --> tests/compile-fail/accumulator_incompatibles.rs:10:22
   |
10 | #[salsa::accumulator(data = MyAcc)]
   |                      ^^^^

error: `db` option not allowed here
  --> tests/compile-fail/accumulator_incompatibles.rs:13:22
   |
13 | #[salsa::accumulator(db = Db)]
   |                      ^^

error: unrecognized option `recover_fn`
  --> tests/compile-fail/accumulator_incompatibles.rs:16:22
   |
16 | #[salsa::accumulator(recover_fn = recover)]
   |                      ^^^^^^^^^^

error: `lru` option not allowed here
  --> tests/compile-fail/accumulator_incompatibles.rs:19:22
   |
19 | #[salsa::accumulator(lru = 12)]
   |                      ^^^

error: `constructor` option not allowed here
  --> tests/compile-fail/accumulator_incompatibles.rs:22:22
   |
22 | #[salsa::accumulator(constructor = Constructor)]
   |                      ^^^^^^^^^^^
